src/libraries/Microsoft.Extensions.Options/tests/Microsoft.Extensions.Options.Tests/OptionsSnapshotTest.cs

```diff
@@ -148,18 +150,29 @@ public void SnapshotOptionsAreCachedPerScope()
                 namedOne = scope.ServiceProvider.GetRequiredService<IOptionsSnapshot<FakeOptions>>().Get("1");
                 Assert.Equal(namedOne, scope.ServiceProvider.GetRequiredService<IOptionsSnapshot<FakeOptions>>().Get("1"));
                 Assert.Equal(2, TestConfigure.ConfigureCount);
+
+                // Reload triggers Configure for the two registered change tokens.
+                config.Reload();
+                Assert.Equal(4, TestConfigure.ConfigureCount);
+
+                // Reload should not affect current scope.
+                var options2 = scope.ServiceProvider.GetRequiredService<IOptionsSnapshot<FakeOptions>>().Value;
+                Assert.Equal(options, options2);
+                var namedOne2 = scope.ServiceProvider.GetRequiredService<IOptionsSnapshot<FakeOptions>>().Get("1");
+                Assert.Equal(namedOne, namedOne2);
             }
             Assert.Equal(1, TestConfigure.CtorCount);
+
+            // Reload should be reflected in a fresh scope.
             using (var scope = factory.CreateScope())
             {
                 var options2 = scope.ServiceProvider.GetRequiredService<IOptionsSnapshot<FakeOptions>>().Value;
                 Assert.NotEqual(options, options2);
-                Assert.Equal(3, TestConfigure.ConfigureCount);
```

Comment:
Why is this assert removed? Can we just change the expected number at this point?

Reply:
That would be another assert against 4 then, it felt a bit redundant.

Reply:
I don't understand how it is redundant, you've done work in between the calls.

Reply:
The thing is it would be caught by the final assert regardless. But as you ask, I'm of the opinion unit tests are for asserting properties that are of importance to the contract and implementation, if either changes so must the tests. The interspersed assert was semantically useful to assert configure would rerun per requested option afresh, however as the changed implementation does not do so anymore it makes sense to just do a sanity check of that fact in the final assert only. But again, I've pushed a commit bringing it back.

Reply:
> The thing is it would be caught by the final assert regardless...The interspersed assert was semantically useful to assert configure would rerun per requested option afresh, however as the changed implementation does not do so anymore it makes sense to just do a sanity check of that fact in the final assert only.

But if there was a bug here, it would be easier to diagnose _when_ the value changed if the test failed here, and not in the final assert.

And if it didn't fail here, but did in the final assert, then you would know it changed between the two asserts.

> But again, I've pushed a commit bringing it back.

I don't see it being brought back.

Reply:
>But if there was a bug here, it would be easier to diagnose when the value changed if the test failed here, and not in the final assert.

OK, diagnostic instead of assertive value, I can agree with that.

> But again, I've pushed a commit bringing it back.

The file was still staged, it's there for real now.